metadata-db-app/src/migrations/064_add_provider_id_to_services_table.clj

```diff
@@ -0,0 +1,54 @@
+(ns migrations.064-add-provider-id-to-services-table
+  "Adds `provider_id` column to services table."
+  (:require
+   [clojure.java.jdbc :as j]
+   [cmr.metadata-db.services.provider-validation :as provider-validation]
+   [config.mdb-migrate-helper :as h]))
+
+(defn- drop-old-constraint
+  []
+  (h/sql
+   "ALTER TABLE METADATA_DB.cmr_services DROP CONSTRAINT services_con_rev"))
+
+(defn- add-new-constraint
+  []
+  (h/sql (str "ALTER TABLE METADATA_DB.cmr_services "
+              "ADD CONSTRAINT services_con_p_n_r "
+              "UNIQUE (native_id, revision_id, provider_id) "
+              "USING INDEX (CREATE UNIQUE INDEX services_idx_p_n_r "
+              "ON cmr_services (provider_id, native_id, revision_id))")))
+
+(defn- drop-new-constraint
+  []
+  (h/sql
+   "ALTER TABLE METADATA_DB.cmr_services DROP CONSTRAINT services_con_p_n_r"))
+
+(defn- re-add-old-constraint
+  []
+  (h/sql (str "ALTER TABLE METADATA_DB.cmr_services "
+              "ADD CONSTRAINT services_con_rev "
+              "UNIQUE (native_id, revision_id) "
+              "USING INDEX (CREATE UNIQUE INDEX services_ucr_i "
+              "ON cmr_services (native_id, revision_id))")))
+
+(defn up
+  "Migrates the database up from version 63 to 64."
+  []
+  (println "migrations.064-add-provider-id-to-services-table up...")
+  ;; At this point in time, UMM-Services support is still being developed, so we
+  ;; don't care about what's in the database right now, and the new field can't
+  ;; be null, so:
+  (h/sql "TRUNCATE TABLE METADATA_DB.cmr_services")
+  (h/sql (format (str "ALTER TABLE METADATA_DB.cmr_services "
+                      "ADD provider_id VARCHAR(%s) NOT NULL")
+                 provider-validation/PROVIDER_ID_MAX_LENGTH))
```

Comment:
We've been bit by using constants/code in our migrations before. While I think it highly unlikely that constant will change our best practice is to paste the value in the migration so that we know no matter when a migration runs it will always use the same value.

Reply:
Ah, good to know.

Reply:
Done.